Modules/mathmodule.c

```diff
@@ -2163,6 +2163,27 @@ math_ldexp_impl(PyObject *module, double x, PyObject *i)
     } else {
         errno = 0;
         r = ldexp(x, (int)exp);
+#if _MSC_VER
+        if (r && r > -DBL_MIN && r < DBL_MIN) {
```

Comment:
This test seems wrong. As I commented in the issue report:

```none
Why can't you just return a 0 result? Because it's not always right. It's possible for x*2**n to yield a result < ulp(TINY), but > ulp(TINY)/2, in which case it should round up to TINY. Like here:

>>> ldexp(0.5000001, -1074) # input a little over ulp(TINY)/2
0.0
>>> ldexpr(0.5000001, -1074) # which my workaround realizes
5e-324
```

But `if (r && ...)` exempts a native 0 result from the correction code.